taskvine/src/manager/vine_manager.c

```diff
@@ -3890,14 +3891,30 @@ static int vine_manager_send_duty_to_worker(struct vine_manager *q, struct vine_
 	vine_manager_send(q,w, "duty %lld %lld\n",  (long long) strlen(name), (long long)t->task_id);
 	link_putlstring(w->link, name, strlen(name), time(0) + q->short_timeout);
 	vine_result_code_t result = start_one_task(q, w, t);
+
+	if(q->txn_logfile)
```

Comment:
Factor this out as a new function in `vine_txn_log.c`.  (That's where all transactions get written)